patterns/pattern-2/src/classification_function/index.py

```diff
@@ -118,14 +123,15 @@ def handler(event, context):
     
     # Check if document processing completely failed or has critical page failures
     if document.status == Status.FAILED or failed_page_exceptions:
+        error_details = '; '.join(str(e) for e in getattr(document, 'errors', []) if e)
```

Comment:
Clarify this change.. Looks unrelated to the Optional Classification and Extraction.